core/lib/multivm/src/interface/traits/vm.rs

```diff
@@ -53,13 +53,14 @@ use crate::interface::types::outputs::{
     BootloaderMemory, CurrentExecutionState, VmExecutionResultAndLogs,
 };
 
+use crate::interface::{FinishedL1Batch, VmMemoryMetrics};
 use crate::vm_latest::HistoryEnabled;
 use crate::HistoryMode;
-use zksync_state::{StoragePtr, WriteStorage};
+use zksync_state::StoragePtr;
 use zksync_types::Transaction;
 use zksync_utils::bytecode::CompressedBytecodeInfo;
 
-pub trait VmInterface<S: WriteStorage, H: HistoryMode> {
+pub trait VmInterface<S, H: HistoryMode> {
```

Comment:
Not sure whether this is applicable in the current case, but it usually makes sense to split such large traits into smaller parts with well-defined boundaries. E.g., in this case methods that don't depend on a `TracerDispatcher` could be extracted to a separate trait that would be used as a super trait for `VmInterface`.

Reply:
it's already splitted to two traits.
I'm not sure i want to introduce more of them..